lib/route/link/macsec.c

```diff
@@ -0,0 +1,571 @@
+{
+	struct macsec_info *info = link->l_info;
+
+	IS_MACSEC_LINK_ASSERT(link);
+
+	info->cipher_suite = cipher_suite;
+
+	return 0;
+}
+
+uint64_t rtnl_link_macsec_get_cipher_suite(struct rtnl_link *link)
+{
+	struct macsec_info *info = link->l_info;
+
+	IS_MACSEC_LINK_ASSERT(link);
+
+	return info->cipher_suite;
+}
+
+int rtnl_link_macsec_set_icv_len(struct rtnl_link *link, uint16_t icv_len)
+{
+	struct macsec_info *info = link->l_info;
+
+	IS_MACSEC_LINK_ASSERT(link);
+
+	info->icv_len = icv_len;
+
+	return 0;
+}
+
+uint16_t rtnl_link_macsec_get_icv_len(struct rtnl_link *link)
+{
+	struct macsec_info *info = link->l_info;
+
+	IS_MACSEC_LINK_ASSERT(link);
+
+	return info->icv_len;
+}
+
+int rtnl_link_macsec_set_protect(struct rtnl_link *link, uint8_t protect)
+{
+	struct macsec_info *info = link->l_info;
+
+	IS_MACSEC_LINK_ASSERT(link);
+
+	info->protect = protect;
+
+	return 0;
+}
+
+uint8_t rtnl_link_macsec_get_protect(struct rtnl_link *link)
+{
+	struct macsec_info *info = link->l_info;
+
+	IS_MACSEC_LINK_ASSERT(link);
+
+	return info->protect;
+}
+
+int rtnl_link_macsec_set_encrypt(struct rtnl_link *link, uint8_t encrypt)
+{
+	struct macsec_info *info = link->l_info;
+
+	IS_MACSEC_LINK_ASSERT(link);
+
+	info->encrypt = encrypt;
+
+	return 0;
+}
+
+uint8_t rtnl_link_macsec_get_encrypt(struct rtnl_link *link)
+{
+	struct macsec_info *info = link->l_info;
+
+	IS_MACSEC_LINK_ASSERT(link);
+
+	return info->encrypt;
+}
+
+int rtnl_link_macsec_set_encoding_sa(struct rtnl_link *link, uint8_t encoding_sa)
+{
+	struct macsec_info *info = link->l_info;
+
+	IS_MACSEC_LINK_ASSERT(link);
+
+	info->encoding_sa = encoding_sa;
+
+	return 0;
+}
+
+uint8_t rtnl_link_macsec_get_encoding_sa(struct rtnl_link *link)
+{
+	struct macsec_info *info = link->l_info;
+
+	IS_MACSEC_LINK_ASSERT(link);
+
+	return info->encoding_sa;
+}
+
+int rtnl_link_macsec_set_validation_type(struct rtnl_link *link, enum macsec_validation_type validate)
+{
+	struct macsec_info *info = link->l_info;
+
+	IS_MACSEC_LINK_ASSERT(link);
+
+	info->validate = validate;
+
+	return 0;
+}
+
+enum macsec_validation_type rtnl_link_macsec_get_validation_type(struct rtnl_link *link)
+{
+	struct macsec_info *info = link->l_info;
+
+	IS_MACSEC_LINK_ASSERT(link);
+
+	return info->validate;
+}
+
+int rtnl_link_macsec_set_replay_protect(struct rtnl_link *link, uint8_t replay_protect)
+{
+	struct macsec_info *info = link->l_info;
+
+	IS_MACSEC_LINK_ASSERT(link);
+
+	info->replay_protect = replay_protect;
+
+	return 0;
+}
+
+uint8_t rtnl_link_macsec_get_replay_protect(struct rtnl_link *link)
+{
+	struct macsec_info *info = link->l_info;
+
+	IS_MACSEC_LINK_ASSERT(link);
+
+	return info->replay_protect;
+}
+
+int rtnl_link_macsec_set_window(struct rtnl_link *link, uint32_t window)
+{
+	struct macsec_info *info = link->l_info;
+
+	IS_MACSEC_LINK_ASSERT(link);
+
+	info->window = window;
+
+	return 0;
+}
+
+uint32_t rtnl_link_macsec_get_window(struct rtnl_link *link)
```

Comment:
I know, there are many getters that follow this pattern. But this gives no way to return an error.

I'd prefer something like
  int rtnl_link_macsec_get_window(struct rtnl_link *link, uint32_t *window);
Like in commit 207dc0352779af46121a83fb4758cdd3155d9001

Also, usually rtnl objects have a mask field so you know which attributes are present or which not. I know it's cumbersome to add, but could we add that?